app/services/p2p_service.py

```
@@ -0,0 +1,539 @@
+        db: Session,
+        *,
+        network_name: str,
+        listen_host: str,
+        listen_port: int,
+        enabled: bool,
+        psk: Optional[str] = None,
+    ) -> P2PNetworkConfig:
+        config = self.get_network_config(db)
+        if config is None:
+            if not psk:
+                raise ValueError("A PSK is required to create the network configuration")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)